kingpin/actors/aws/ecs.py

```diff
@@ -0,0 +1,410 @@
+            raise gen.Return((task_definition['family'], 1))
+
+        response = yield self.thread(
+            self.ecs_conn.register_task_definition, **task_definition)
+
+        # Parse data from the server's response.
+        task_definition = response['taskDefinition']
+        family = task_definition['family']
+        revision = task_definition['revision']
+
+        self.log.info('Task definition {0}:{1} registered'.format(
+            family, revision))
+        raise gen.Return((family, revision))
+
+    @gen.coroutine
+    @dry('Would run task {0}:{1}')
+    def _run_task(self, family, revision):
+        """Runs a task on ECS given a task definition's family and revision.
+
+        A task can result in multiple running tasks,
+        depending on count and sidekick tasks.
+
+        Args:
+            family: Task family string
+            revision: Task revision string
+
+        Returns:
+            list: task ARNs.
+        """
+        response = yield self.thread(
+            self.ecs_conn.run_task,
+            cluster=self.option('cluster'),
+            taskDefinition='{0}:{1}'.format(family, revision),
+            count=self.option('count'))
+        self._handle_failures(response['failures'])
+        tasks = [t['taskArn'] for t in response['tasks']]
+        raise gen.Return(tasks)
+
+    @gen.coroutine
+    @dry('Would wait for tasks')
+    def _wait_for_tasks(self, tasks):
+        """Wait for tasks to stop on ECS.
+
+        If a task stops with a non-zero exit code,
+        this will raise an exception.
+
+        Args:
+            tasks: list of task ARNs to wait for.
+        """
+        while True:
+            done = yield self._tasks_done(tasks)
+            if done:
+                break
+            yield gen.sleep(5)
+
+    @gen.coroutine
+    def _tasks_done(self, tasks):
+        """Checks if tasks are done.
+
+        Args:
+            tasks: list of task ARNs to check.
+
+        Returns:
+            A boolean indicating whether all tasks are done.
+        """
+        response = yield self.thread(
+            self.ecs_conn.describe_tasks,
+            cluster=self.option('cluster'),
+            tasks=tasks)
+
+        self._handle_failures(response['failures'])
+
+        containers = _get_containers_from_tasks(
+            task_list=response['tasks'])
+
+        total_count = len(containers)
+        stopped_count = 0
+
+        for container in containers:
+            if container['lastStatus'] == 'STOPPED':
+                stopped_count += 1
+            exit_code = container.get('exitCode', None)
+            if exit_code and exit_code != 0:
+                self.log.error('Task {0} exited with code {1}'.format(
+                    container['taskArn'], exit_code))
+                raise exceptions.UnrecoverableActorFailure()
+
+        if stopped_count == total_count:
+            self.log.info('All {0} tasks stopped'.format(total_count))
+            raise gen.Return(True)
```

Comment:
Add a new line after this `if` block. These two steps aren't connected logically